Make parameter and objective classes slotted

AdrianSosic opened the pull request:
Now that `attrs==24.1.0` supports `cached_property` with slotted classes, we can finally enable slots for our parameter and objective classes.

AdrianSosic commented:
Note: there was another [issue](https://github.com/python-attrs/attrs/issues/1220) which messed with our subclass detection mechanism, which has already be resolved but the fix has not yet been released. However, the checks should automatically pass once attrs `24.1.0` is available.

AVHopp reviewed (approved):
I guess there is not really a need for a proper and full review, this is just some maintenance right?

Scienfitz reviewed (commented):
I have no objections, but this PR should only be merged once the new attrs version is available right?

Scienfitz commented on pyproject.toml:
```diff
@@ -29,7 +29,7 @@ keywords = [
 dynamic = ['version']
 dependencies = [
-    "attrs>=22.2.0",
+    "attrs>=24.1.0",
```
I have no objections, but this PR should only be merged once the new attrs version is available on pip right?

AdrianSosic replied:
correct 👍🏼

AdrianSosic commented:
@AVHopp @Scienfitz, FYI: I rebased and added one more commit since in the meantime we also had to make our `Objective` classes slotted ... a change that can be similarly reverted by this PR once the attrs support has been released

AVHopp commented on CHANGELOG.md:
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 
+## [Unreleased]
+### Fixed
+- Parameter and objective classes are now slotted due to `cached_property` attrs support
```
@AdrianSosic what is preventing us from rebasing/merging this? Is there anything still open?

AdrianSosic replied:
Will show you in a minute by linking to the corresponding issue (and closing this thread). Will need to put on hold again until resolved =/

AdrianSosic commented:
Again on hold until the following blocking issue gets resolved:
https://github.com/python-attrs/attrs/issues/1325

AdrianSosic commented:
Not happening much on the `attrs` side at the moment but potentially the problem becomes obsolete when switching to `mkdocs`, so let's keep it idling for a few more weeks...

AdrianSosic commented:
> @AdrianSosic - "a few more weeks" is over by half a year, any updates here?

Nothing I can do at the moment. I know that there is something happing due to changes in Python 3.14, but unclear if it will resolve the issue. And workarounds/fixes from our end would require waiting for the new docs. I'll close the issue now and will reconsider once there is a reason to do so
